mobile/src/reducers/mobile.js

```diff
@@ -1,25 +1,35 @@
-import { SETUP, SET_URL, SET_STATE, ERROR } from '../actions'
+import {
+  SETUP, SET_URL, SET_STATE, ERROR, UPDATE_SETTINGS,
+  SHOW_UNLINK_CONFIRMATION, HIDE_UNLINK_CONFIRMATION, UNLINK
+} from '../actions'
 
 const initialState = {
   isSetup: false,
   serverUrl: '',
   settings: {
-    backupImages: false
+    backupImages: false,
+    displayUnlinkConfirmation: false
   }
 }
 
 export const mobile = (state = initialState, action) => {
   switch (action.type) {
     case SET_STATE:
-      return action.state
+      return Object.assign({}, initialState, action.state)
+    case UNLINK:
+      return Object.assign({}, initialState)
```

Comment:
and as `state` default value is `initialState`, we may use `state` instead, to stay pure.